components/authorization/backend/src/main/kotlin/rights/OrganizationPermission.kt

```diff
@@ -49,3 +49,12 @@ enum class OrganizationPermission {
     /** Permission to delete the [Organization]. */
     DELETE
 }
+
+/**
+ * The set of permissions required by the role to read an organization. (This is defined here to avoid circular
+ * dependencies, as it is referenced by multiple role classes.)
+ */
```

Comment:
Are you sure that is required? I think the compiler can handle this kind of circular dependencies, I must made a small test that compiles:

```kotlin
enum class OrgRoles(val orgPermissions: List<String>, val prodPermissions: List<String>) {
    READER(
        orgPermissions = listOf("READ"),
        prodPermissions = ProdRoles.READER.prodPermissions,
    )
}

enum class ProdRoles(val orgPermissions: List<String>, val prodPermissions: List<String>) {
    READER(
        orgPermissions = OrgRoles.READER.orgPermissions,
        prodPermissions = listOf("READ"),
    )
}
```